praetorian_cli/sdk/chariot.py

```
@@ -70,16 +70,16 @@ def __init__(self, keychain: Keychain, proxy: str=''):
+    def chariot_request(self, method: str, url: str, headers: dict | None = None, **kwargs) -> requests.Response:
         """
-        Centralized wrapper around requests.request. Takes care of proxy and 
+        Centralized wrapper around requests.request. Takes care of proxy and
         supplies the authentication headers
         """
         if self.proxy:
             kwargs['proxies'] = {'http': self.proxy, 'https': self.proxy}
             kwargs['verify'] = False
 
-        return requests.request(method, url, headers=(headers | self.keychain.headers()), **kwargs)
+        return requests.request(method, url, headers=((headers or {}) | self.keychain.headers()), **kwargs)
```

Ruff (0.15.15)

[error] Probable use of `requests` call without timeout

(S113)